app/src/main/java/org/oppia/app/drawer/NavigationDrawerFragmentPresenter.kt

```diff
@@ -0,0 +1,102 @@
+package org.oppia.app.drawer
+
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.ActionBarDrawerToggle
+import androidx.appcompat.widget.Toolbar
+import androidx.drawerlayout.widget.DrawerLayout
+import androidx.fragment.app.Fragment
+import com.google.android.material.navigation.NavigationView
+import org.oppia.app.R
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.help.HelpActivity
+import org.oppia.app.home.HomeActivity
+import javax.inject.Inject
+
+/** The presenter for [NavigationDrawerFragment]. */
+@FragmentScope
+class NavigationDrawerFragmentPresenter @Inject constructor(
+  private val fragment: Fragment
+) : NavigationView.OnNavigationItemSelectedListener {
+  private lateinit var navView: NavigationView
+  private lateinit var drawerToggle: ActionBarDrawerToggle
+  private lateinit var drawerLayout: DrawerLayout
+  private var previousMenuItemId: Int? = null
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
+    val view: View? = inflater.inflate(R.layout.fragment_drawer, container, false)
+    navView = view!!.findViewById(R.id.nav_view)
+    navView.setNavigationItemSelectedListener(this)
+    fragment.setHasOptionsMenu(true)
+    return view
+  }
+
+  private fun openActivityByMenuItemId(menuItemId: Int) {
+    if (previousMenuItemId != menuItemId && menuItemId != 0) {
+      var intent = Intent(fragment.activity, HomeActivity::class.java)
+      when (menuItemId) {
+        R.id.nav_home -> {
+          intent = Intent(fragment.activity, HomeActivity::class.java)
+        }
+        R.id.nav_help -> {
+          intent = Intent(fragment.activity, HelpActivity::class.java)
+        }
+      }
+      fragment.activity!!.startActivity(intent)
+      fragment.activity!!.finish()
+    } else {
+      drawerLayout.closeDrawers()
+    }
+  }
+
+  /** This function contains the DrawerListener and also set the drawer toggle. */
+  fun setUpDrawer(drawerLayout: DrawerLayout, toolbar: Toolbar, menuItemId: Int) {
+    when (menuItemId) {
+      R.id.nav_home -> {
```

Comment:
Could we use an enum for this? It seems like it'd be nice to generalize the associations between the IDs and the item indexes in a single location to simplify the two ``when`` statements in this presenter.

Reply:
Resolved https://gist.github.com/nikitamarysolomanpvt/41198682f8a96deffac9b55bcf526d10